vault/barrier_view.go

```diff
@@ -19,7 +19,7 @@ import (
 // BarrierView implements logical.Storage so it can be passed in as the
 // durable storage mechanism for logical views.
 type BarrierView struct {
-	storage         *logical.StorageView
+	storage         logical.StorageView
```

Comment:
@cipherboy : I didn't understand why we have to replace the pointer by the structure itself.

Reply:
We didn't :-) 

In `sdk/logical/storage_view.go` we renamed `StorageView` (the struct) to `storageView` (note lowercase S, so it is not exported). `StorageView` is now an interface (in a two parts -- the `View` portion called `StorageViewCore` and the `Storage` portion), implemented by `storageView` with pointer receivers.

So, this change replaces a pointer-to-struct with an interface, which is mostly equivalent to pointer-to-struct, except that it hides the internal implementation details (which weren't directly used anyways).

Reply:
@cipherboy : OK, thank you!